### PR TITLE
finishing touches for artist header view

### PR DESCRIPTION
**Pull request by TOMalley104**
- pass SPTArtist to ArtistHeaderView
- add pod for loading images 
- update Header View for set SPTArtist (including images)
- fix how xib is loaded and set up (there was an extra view layer in there, used file's owner)
- updates constraints so that the image view isn't gigantic when given a 20k by 20k image 
- add followers label and follow button

### CODE REVIEW

File: spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift [hunk 1]
```diff
@@ -7,35 +7,47 @@
...
     @IBOutlet weak var profileImageView: UIImageView!
     @IBOutlet weak var headerImageView: UIImageView!
     @IBOutlet weak var blurImageView: UIImageView!
     @IBOutlet weak var blurContainerView: UIView!
+    @IBOutlet weak var blurHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var followButton: UIButton! {
+        didSet {
+            self.followButton.layer.borderColor = UIColor.lightGrayColor().CGColor
+            self.followButton.layer.borderWidth = 0.5
+            self.followButton.layer.cornerRadius = self.followButton.ip_halfHeight
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistHeaderView.swift]: we really should make ip_roundView :/

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 2]
```diff
@@ -12,14 +12,10 @@ import FontAwesomeKit
...
-        didSet {
-            if let artist = self.artist {
-                self.updateViews(withArtist: artist)
-            }
-        }
-    }
-    @IBOutlet weak var artistHeaderView: ArtistHeaderView!
+    var artist: SPTArtist?
+    var artistHeaderView: ArtistHeaderView = {
+        return NSBundle.mainBundle().loadNibNamed(ArtistHeaderView.ip_nibName, owner: nil, options: nil).first as! ArtistHeaderView
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: this was just so you understood how this worked - use there should be an ip_loadXX function for this

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 3]
```diff
@@ -117,8 +115,25 @@ class ArtistViewController: UIViewController, UIScrollViewDelegate, UITableViewD
...
+        bottomToShufflePlayCenterY.priority = 750
+        bottomToShufflePlayCenterY.active = true
+
+        self.artistHeaderView.artist = artist
+        self.view.sendSubviewToBack(self.artistHeaderView)
+    }
+
     func setupMoreBarButton() {
-        let iconSize = 40 as CGFloat
+        let iconSize = 40.0 as CGFloat
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: do you still need as CGFloat after the 40.0?

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 4]
```diff
@@ -114,28 +113,29 @@ class ArtistViewController: UIViewController, UIScrollViewDelegate, UITableViewD
...
+        bottomToShufflePlayCenterY.priority = 750
+        bottomToShufflePlayCenterY.active = true
+
+        self.artistHeaderView.artist = artist
+        self.view.sendSubviewToBack(self.artistHeaderView)
     }
 
     func setupMoreBarButton() {
-        let iconSize = 40 as CGFloat
+        let iconSize = 40.0 as CGFloat
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: this comment here still stands

File: spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+class ArtistSearchTableViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, SPTAuthViewDelegate {
+
+    @IBOutlet weak var searchBar: UISearchBar!
+    @IBOutlet weak var logInBarButton: UIBarButtonItem!
+    @IBOutlet weak var tableView: UITableView!
+
+    var session: SPTSession? {
+        didSet {
+            if let session = self.session {
+                self.logInBarButton.enabled = !session.isValid() // FIXME: isValid() is lying
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Search/ArtistSearchTableViewController.swift]: how to fix 'isValid() is lying'?

**Comment by brightredchilli** [comment]
actually, rebase this and resolve conflicts, and I will review one more time before giving a ship it

**Comment by TOMalley104** [comment]
@brightredchilli rebased and ready for review 👍

File: spotifyMini/ViewControllers/Artist/ArtistViewController.swift [hunk 6]
```diff
@@ -97,17 +93,18 @@ class ArtistViewController: UIViewController, UIScrollViewDelegate, UITableViewD
     }
 
     func animateHeaderView(withOffset yOffset: CGFloat) {
-        let userIsPullingDown = yOffset < 0
+        let userIsPullingDown = yOffset <= 0
         if userIsPullingDown {
             let affectedAlpha = 1.5 + (yOffset/100.0)
             self.artistHeaderView.blurAlpha = affectedAlpha
+            self.artistHeaderView.blurHeightConstraint.constant = 325 + -yOffset
```
Comment (brightredchilli) [inline spotifyMini/ViewControllers/Artist/ArtistViewController.swift]: + - is really just '-' :/

**Comment by brightredchilli** [comment]
one more small comment, then :ship:

**Pull request merged**